src/PowerShellEditorServices/Services/PowerShellContext/Session/PSReadLinePromptContext.cs

```diff
@@ -129,7 +132,7 @@ public async Task<string> InvokeReadLineAsync(bool isCommandLine, CancellationTo
             }
 
             var readLineCommand = new PSCommand()
-                .AddCommand("__Invoke-ReadLineForEditorServices")
+                .AddCommand(new CmdletInfo("__Invoke-ReadLineForEditorServices", s_lazyReadLineCmdletType.Value))
```

Comment:
Maybe cache this `CmdletInfo` since it'll be used a ton.  Other one probably doesn't need to be tho.

A simple thread unsafe cache like:

```csharp
private static CmdletInfo s_cmdlet;

s_cmdlet ??= new CmdletInfo(etc);
AddCommand(s_cmdlet);
```

In the event of a race condition it'll just make an extra one which is nbd

Reply:
Actually I might as well make that the lazy part

Reply:
I put those into the lazy part since we only care about the cmdletinfo